examples/SampleApp/src/hooks/useClientNotificationsToastHandler.ts

```diff
@@ -0,0 +1,39 @@
+import type { Notification } from 'stream-chat';
+import { useClientNotifications } from 'stream-chat-react-native';
+
+import { useEffect, useMemo, useRef } from 'react';
+import { useToastState } from './useToastState';
+
+export const usePreviousNotifications = (notifications: Notification[]) => {
+  const prevNotifications = useRef<Notification[]>(notifications);
+
+  const difference = useMemo(() => {
+    const prevIds = new Set(prevNotifications.current.map((notification) => notification.id));
+    const newIds = new Set(notifications.map((notification) => notification.id));
+    return {
+      added: notifications.filter((notification) => !prevIds.has(notification.id)),
+      removed: prevNotifications.current.filter((notification) => !newIds.has(notification.id)),
+    };
+  }, [notifications]);
+
+  useEffect(() => {
+    prevNotifications.current = notifications;
```

Comment:
This entire `useEffect` can be replaced with:

```
prevNotifications.current = notifications;
```

This is because:

- `notifications` is not going to change referentially unless the actual array changes
- Changing the `ref` on every render is not going to trigger any rerenders, so that's also fine